app/src/main/java/net/gsantner/markor/frontend/textview/HighlightingEditor.java

```diff
@@ -95,6 +107,66 @@ public void afterTextChanged(final Editable s) {
         setEmojiCompatEnabled(false);
     }
 
+    @Override
+    protected void onFinishInflate() {
+        super.onFinishInflate();
+        _defaultPaddingLeft = getPaddingLeft();
+    }
+
+    @Override
+    protected void onDraw(Canvas canvas) {
+        super.onDraw(canvas);
+
+        // If line numbers enabled
+        if (_nuEnabled) {
+            drawLineNumbers(canvas);
+        } else if (getPaddingLeft() != _defaultPaddingLeft) {
+            // Reset padding without line numbers fence
+            setPadding(_defaultPaddingLeft, getPaddingTop(), getPaddingRight(), getPaddingBottom());
+        }
+    }
+
+    private void drawLineNumbers(Canvas canvas) {
+        final int firstBaselineToTopHeight = getPaddingTop() - getPaint().getFontMetricsInt().top;
+        int number = 1;
+
+        _paint.setColor(Color.GRAY);
+        // Draw first line number
+        canvas.drawText(String.valueOf(number), LINE_NUMBERS_PADDING_LEFT, firstBaselineToTopHeight, _paint);
+        // Draw others line number
+        float y;
+        final float offsetY = firstBaselineToTopHeight - getTextSize() * 0.14f;
```

Comment:
Why 0.14?

Reply:
line spacing I guess

Reply:
It is a vertical downward offset with using getLineTop(i), 0.14 was got by trial.